tests/poly/runner/lower_mul_tests.toml

```diff
@@ -0,0 +1,6 @@
+[[test]]
+ideal = "1 + x**12"
+cmod = 4294967296  # 2**32
+p0 = "1 + x**10"
+p1 = "1 + x**11"
+container_type = "i32"
```

Comment:
will there be some error if container_type is larger than bit size of cmod? (https://github.com/google/heir/blob/8ca202760fed9f966a3c10ac6ff990f6af40f5d4/lib/Dialect/Poly/IR/PolyOps.cpp#L43-L50) or are the TOML files expected to be generated so it's always true? (is the reason this arg isn't explicit because you want to test fromtensor op's extension? if so, i feel like it's not necessarily useful, since these are testing the poly op lowerings correctness)

Reply:
The container_type is meant to be the element type of the tensor output by the lowered poly.mul. I felt it was too hard to compute that value at test generation time, or rather, it duplicates too much of the logic of the thing it's testing. So it's assumed to be set correctly by the user. If it doesn't exactly match the output of the lowered poly mul, it will be a verifier error at parse time, because the templated ops will be doing a "trunci" or "extsi" with a type signature that doesn't match the type of the input. So it seems pretty hard to silently get wrong.